src/Lecture4.hs

```diff
@@ -183,9 +238,25 @@ The 'Stats' data type has multiple fields. All these fields have
 instance for the 'Stats' type itself.
 -}
 
-instance Semigroup Stats where
+strictSemiGroupMaybe :: Semigroup a => Maybe a -> Maybe a -> Maybe a
+strictSemiGroupMaybe Nothing a = a
+strictSemiGroupMaybe a Nothing = a
+strictSemiGroupMaybe (Just !a) (Just !b) = Just (a <> b)
```

Comment:
That's absolutely correct implementation 👍🏻 
You can write it a bit shorter though by pattern matching only on the final result:

```suggestion
strictSemiGroupMaybe a b = case a <> b of
    Nothing -> Nothing
    Just !x -> Just x
```

This implementation also will be lazier in a good way 🦥